File: 205-isomorphic-strings/isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        map1 = [0] * 256
        map2 = [0] * 256

        for i in range(len(s)):
            if map1[ord(s[i])] != map2[ord(t[i])]:
                return False
            map1[ord(s[i])] = map2[ord(t[i])] = i + 1

        return True
    

File: 205-isomorphic-strings/isomorphic-strings.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isIsomorphic = function(s, t) {
    if (s.length !== t.length) return false;

    const map1 = Array(256).fill(0);
    const map2 = Array(256).fill(0);

    for (let i = 0; i < s.length; i++) {
        if (map1[s.charCodeAt(i)] !== map2[t.charCodeAt(i)]) return false;
        map1[s.charCodeAt(i)] = map2[t.charCodeAt(i)] = i + 1;
    }

    return true;
};
